lib/psych.rb

```diff
@@ -498,14 +538,20 @@ def self.to_json object
   #   end
   #   list # => ['foo', 'bar']
   #
-  def self.load_stream yaml, filename = nil
-    if block_given?
-      parse_stream(yaml, filename) do |node|
-        yield node.to_ruby
-      end
-    else
-      parse_stream(yaml, filename).children.map { |child| child.to_ruby }
+  def self.load_stream yaml, legacy_filename = NOT_GIVEN, filename: nil, fallback: false
```

Comment:
It‘s not clear to me under what conditions the fallback will be returned here.

~~~ruby
Psych.load_stream "", fallback: 42  # => []
# neither the specified value nor the default (false) is used in this case

# whereas
Psych.load "", fallback: 42  # => 42
~~~

Reply:
👀 Looking into this, was away on Holiday sorry for late response

Reply:
Just came back to it.

Figured out the discrepancy with the `fallback` when calling `load_stream` (return: `[]`)

Basically, `parse_stream` (called in `load_stream`) returns an empty array for an empty YAML value. Thus, in this case, the fallback isn't called because it uses the empty array from `parse_stream` as the result.

I'll investigate to see whether this is a normal case, a unique case and/or a bug. I'll see how to proceed and update as necessary.

NOTE: While looking into this, I noticed a thing that should reduce the amount of code and complexity, I'll fiddle and see if I can figure something that's a bit cleaner, simpler and without changes to the outcomes (no need to bump major) --> will do so in another PR